account_payment_other_company/models/account_payment.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountPayment(models.Model):
+    _inherit = 'account.payment'
+
+    other_journal_id = fields.Many2one('account.journal',
+                                       string='Paid By')
+    hide_other_journal = fields.Boolean(default=True,
+                                        invisible=True)
+
+    @api.multi
+    def action_validate_invoice_payment(self):
+        res = super().action_validate_invoice_payment()
+        if not self.hide_other_journal:
+            account_move = self.env['account.move']
+            vals = self._prepare_other_payment_values()
+            move = account_move.sudo().create(vals)
+            move.partner_id = self.partner_id
+            move.sudo().post()
+        return res
+
+    @api.onchange('journal_id', 'payment_type')
+    def _onchange_hide_other_journal(self):
+        if self.journal_id.id == self.\
+                company_id.due_fromto_payment_journal_id.id and \
+                self.payment_type == 'outbound':
+            self.hide_other_journal = False
+        else:
+            self.hide_other_journal = True
```

Comment:
Ideomatic suggestion: when assigning boolean values, we can usually avoid the explicit `if`
```
        self.hide_other_journal = not (
            self.journal_id.id == self.company_id.due_fromto_payment_journal_id.id and
            self.payment_type == 'outbound')
```